language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/connector/BallerinaConnectorService.java

```diff
@@ -151,50 +136,23 @@ public void init(LanguageServer langServer, WorkspaceManager workspaceManager,
      *
      * @param filePath file path
      * @param detailed detailed connector out put or not
+     * @param query    search query to filter connector list
      * @return connector list
      * @throws IOException
      */
-    private List<Connector> fetchLocalConnectors(Path filePath, boolean detailed) {
-        ProjectEnvironmentBuilder defaultBuilder = ProjectEnvironmentBuilder.getDefaultBuilder();
-        defaultBuilder.addCompilationCacheFactory(TempDirCompilationCache::from);
-        Project balaProject = ProjectLoader.loadProject(filePath, defaultBuilder);
+    private List<Connector> fetchLocalConnectors(Path filePath, boolean detailed, String query) {
+        Optional<Project> project = workspaceManager.project(filePath);
         List<Connector> connectors = new ArrayList<>();
         try {
-            connectors = ConnectorGenerator.getProjectConnectors(balaProject, detailed);
+            if (project.isPresent()) {
+                connectors = ConnectorGenerator.getProjectConnectors(project.get(), detailed, query);
```

Comment:
Since the array has been already initialized, you don't need to re-assign. You can use the `addAll` API.

Reply:
Thanks, updated